core/src/main/java/com/linecorp/armeria/common/RequestContextExporter.java

```diff
@@ -114,7 +123,31 @@
         }
     }
 
-    void export(Map<String, String> out, RequestContext ctx, RequestLog log) {
+    /**
+     * Returns a {@link Map} whose key is an export key set through {@code add*()} in
+     * {@link RequestContextExporterBuilder} and value is extracted from {@link RequestContext}
+     * Note that: this method returns {@code null} if current {@link RequestContext} is {@code null}.
+     */
+    @Nullable
+    public Map<String, String> export() {
+        final RequestContext ctx = RequestContext.currentOrNull();
+        if (ctx != null) {
+            final State state = state(ctx);
+            final RequestLog log = ctx.log();
+            final Set<RequestLogAvailability> availabilities = log.availabilities();
+
+            // Note: This equality check is extremely fast.
+            //       See RequestLogAvailabilitySet for more information.
+            if (!availabilities.equals(state.availabilities)) {
+                state.availabilities = availabilities;
+                export(state, ctx, log);
```

Comment:
I think after moving this class under `common` and being public class, a user could call this method any time. If we add `RequestLogAvailability.COMPLETE` then we need description in documentation or throwing an exception?

Reply:
> Don't we need to call only when `RequestLogAvailability` is `complete` because `exportAttributes()` and others can be called multiple times?

I think this is intentional, because context properties must be exported partially even if the request is not complete. Otherwise, log messages of the requests being processed will not have any MDC properties set.

Reply:
Thanks for the explanation. 😄

Reply:
I've learned too. 🙇‍♂️